bluemira/equilibria/fem_fixed_boundary/equilibrium.py

```diff
@@ -0,0 +1,256 @@
+        delta_l = builder_plasma.params.get_param("delta_l")
+
+        kappa = (kappa_u + kappa_l) / 2
+        delta = (delta_u + delta_l) / 2
+
+        if verbose:
+            print(f"kappa_u: {kappa_u}, delta_u: {delta_u}")
+            print(f"kappa_l: {kappa_l}, delta_l: {delta_l}")
+            print(f"kappa: {kappa}, delta: {delta}")
+            print(f"volume: {plasma_volume}")
+
+        # initialize plasmod solver
+        # - V_p is set equal to plasma volume
+        plasmod_options["params"].set_parameter("V_p", plasma_volume, "m^3", source)
+
+        plasmod_options["params"].set_parameter("kappa", kappa, "dimensionless", source)
+        plasmod_options["params"].set_parameter("delta", delta, "dimensionless", source)
+        plasmod_options["params"].set_parameter(
+            "kappa_95", kappa_95, "dimensionless", source
+        )
+        plasmod_options["params"].set_parameter(
+            "delta_95", delta_95, "dimensionless", source
+        )
+
+        plasmod_solver = PlasmodTransportSolver(
+            params=plasmod_options["params"],
+            build_config=plasmod_options["build_config"],
+        )
+        plasmod_solver.execute()
+
+        if plot:
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.pprime(plasmod_solver.x), "pprime", "-"
+            )
+            plot_profile(
+                plasmod_solver.x, plasmod_solver.ffprime(plasmod_solver.x), "ffrime", "-"
+            )
+
+        # generate mesh for the Grad-Shafranov solver
+        plasma.shape.boundary[0].mesh_options = {
+            "lcar": lcar_mesh,
+            "physical_group": "lcfs",
+        }
+        plasma.shape.mesh_options = {
+            "lcar": lcar_mesh,
+            "physical_group": "plasma_face",
+        }
+
+        m = meshing.Mesh()
+        m(plasma)
+
+        msh_to_xdmf("Mesh.msh", dimensions=(0, 2), directory=".", verbose=verbose)
+
+        mesh = import_mesh(
+            "Mesh",
+            directory=".",
+            subdomains=True,
+        )[0]
+
+        # initialize the Grad-Shafranov solver
+        p_order = gs_options["p_order"]
+        gs_solver = FemGradShafranovFixedBoundary(mesh, p_order=p_order)
+
+        bluemira_print("Solving fixed boundary Grad-Shafranov...")
+
+        gs_solver.solve(
+            plasmod_solver.pprime,
+            plasmod_solver.ffprime,
+            plasmod_solver.I_p,
+            iter_err_max=gs_options["iter_err_max"],
+            max_iter=gs_options["max_iter"],
+            relaxation=gs_options["relaxation"],
+            plot=gs_options["plot"],
+        )
+
+        _, kappa_95, delta_95 = calculate_plasma_shape_params(
+            gs_solver.psi_norm_2d,
+            mesh,
+            np.sqrt(0.95),
+        )
+
+        # calculate the iteration error
+        err_delta = abs(delta_95 - delta95_t) / delta95_t
+        err_kappa = abs(kappa_95 - kappa95_t) / kappa95_t
+        iter_err = max(err_delta, err_kappa)
```

Comment:
In future, we could use a `ConvergenceCriterion` (as in equilibria/solve.py), but it is not necessary for this PR.

In future, we could use a more generic version of the `PicardIterator` (as in equilibria/solve.py) but right now this functionality only makes sense for free boundary equilibria with coils (as they are used to stabilise the problem, in crude terms) - none of which is necessary for this Picard iteration.